src/app/components/table.tsx

```diff
@@ -0,0 +1,177 @@
+import { useMemo } from 'react';
+
+type DataNode = LeafNode | ParentNode;
+type LeafNode = {
+  name: string;
+  value: number;
+};
+type ParentNode = {
+  name: string;
+  children: DataNode[];
+};
+
+export default function Table({ data }: { data: DataNode }) {
+  const total = useMemo<number>(() => getCount(data), [data]);
+  if (hasGrandChildren(data)) {
+    return (
+      <table>
+        <thead>
+          <tr>
+            <th>Version</th>
+            <th>Count</th>
+            <th>
+              Percentage of{' '}
```

Comment:
Is this the standard way to do a space at the end of a line in JSX? I don't know anything better but it seems rough

Reply:
Yeah I don't know of anything better either - fwiw this is what prettier formats to when I try to put it all in one line